lib/adhearsion/call_controller/dial.rb

```diff
@@ -169,6 +173,8 @@ def prep_calls
             end
 
             new_call.on_answer do |event|
+              next unless @call.alive? && @call.active?
+
```

Comment:
Thank you for this!  I love the spirit of this change.

However, I think there are activities here that are _not_ dependent on `@call` and _are_ important to still perform.  As such, would you mind doing the following instead of invoking `next`? (sorry in advance for this not being as pretty/DRY):
* tuck the #on_joined and #on_unjoined below in one big fat conditional
* in the `if new_call.alive? && new_call.active? && status.result != :answer` conditional, evaluate `@call alive&active` in that if statement as well